lib/ocpp/v201/functional_blocks/provisioning.cpp

```diff
@@ -0,0 +1,713 @@
+}
+
+void Provisioning::stop_bootnotification_timer() {
+    this->boot_notification_timer.stop();
+}
+
+void Provisioning::on_variable_changed(const SetVariableData& set_variable_data) {
+    this->handle_variable_changed(set_variable_data);
+}
+
+std::vector<GetVariableResult>
+Provisioning::get_variables(const std::vector<GetVariableData>& get_variable_data_vector) {
+    std::vector<GetVariableResult> response;
+    for (const auto& get_variable_data : get_variable_data_vector) {
+        GetVariableResult get_variable_result;
+        get_variable_result.component = get_variable_data.component;
+        get_variable_result.variable = get_variable_data.variable;
+        get_variable_result.attributeType = get_variable_data.attributeType.value_or(AttributeEnum::Actual);
+        const auto request_value_response = this->device_model.request_value<std::string>(
+            get_variable_data.component, get_variable_data.variable,
+            get_variable_data.attributeType.value_or(AttributeEnum::Actual));
+        if (request_value_response.status == GetVariableStatusEnum::Accepted and
+            request_value_response.value.has_value()) {
+            get_variable_result.attributeValue = request_value_response.value.value();
+        }
+        get_variable_result.attributeStatus = request_value_response.status;
+        response.push_back(get_variable_result);
+    }
+    return response;
+}
+
+std::map<SetVariableData, SetVariableResult>
+Provisioning::set_variables(const std::vector<SetVariableData>& set_variable_data_vector, const std::string& source) {
+    // set variables and allow setting of ReadOnly variables
+    const auto response = this->set_variables_internal(set_variable_data_vector, source, true);
+    this->handle_variables_changed(response);
+    return response;
+}
+
+void Provisioning::notify_report_req(const int request_id, const std::vector<ReportData>& report_data) {
+    NotifyReportRequest req;
+    req.requestId = request_id;
+    req.seqNo = 0;
+    req.generatedAt = ocpp::DateTime();
+    req.reportData.emplace(report_data);
+    req.tbc = false;
+
+    if (report_data.size() <= 1) {
+        ocpp::Call<NotifyReportRequest> call(req);
+        this->message_dispatcher.dispatch_call(call);
+    } else {
+        NotifyReportRequestsSplitter splitter{
+            req,
+            this->device_model.get_optional_value<size_t>(ControllerComponentVariables::MaxMessageSize)
+                .value_or(DEFAULT_MAX_MESSAGE_SIZE),
+            []() { return ocpp::create_message_id(); }};
+        for (const auto& msg : splitter.create_call_payloads()) {
+            this->message_queue.push_call(msg);
+        }
+    }
+}
+
+void Provisioning::handle_boot_notification_response(CallResult<BootNotificationResponse> call_result) {
+    // TODO(piet): B01.FR.06
+    // TODO(piet): B01.FR.07
+    // TODO(piet): B01.FR.08
+    // TODO(piet): B01.FR.09
+    // TODO(piet): B01.FR.13
```

Comment:
are now all implemented. B01.FR.13 is missing in the status document, you can add it there as well.
```suggestion
```